test/arrays/ferguson/semantic-examples/1.5-return-alias-slice-global.chpl

```diff
@@ -8,15 +8,15 @@ proc f() {
   return slice;
 }
 
-proc g(x) {
+proc g(in x) {
   x[2] = one;
 }
```

Comment:
I think the switch to in-intent takes us away from answering the question at the bottom of this test. How about er replace `g(f())` with:
```chpl
f()[2] = one;
```

P.S. I am puzzled that my replacement works. However I only tried it on main, not sure whether it will work on your branch.

If we cannot make this and the subsequent tests answer the question they aim to, I do not see the value of keeping them.

Maybe check with @mppf ?

Reply:
I think a temporary `ref` variable for the result of `f` will best preserve the intentions of the test

Reply:
Thinking about these more, I think `var` actually does capture it best, since `f` is returning by value. The reason these temporaries are added is that there are lvalue errors since we changes how that works for arrays recently, which I think has a bad reaction with ref-maybe-const